src/pages/billing/List.js

```diff
@@ -360,12 +360,20 @@ export const BillingList = ({ queryParams: { selectedName } }) => {
               p(['It may not exist, or you may not have access to it.'])
             ])
           ])],
-        [selectedName && hasBillingProjects, () => h(ProjectDetail, {
-          key: selectedName,
-          project: _.find({ projectName: selectedName }, billingProjects),
-          billingAccounts,
-          authorizeAndLoadAccounts
-        })],
+        [selectedName && hasBillingProjects, () => {
+          const index = _.findIndex({ projectName: selectedName }, billingProjects)
+          return h(ProjectDetail, {
+            key: selectedName,
+            billingProject: billingProjects[index],
+            billingAccounts,
+            authorizeAndLoadAccounts,
+            updateProjectAndReload: async () => {
+              const projects = billingProjects.slice()
+              projects[index] = await Ajax(signal).Billing.billingProject(selectedName)
+              setBillingProjects(projects)
+            }
+          })
```

Comment:
I had added a suggestion in here yesterday, not sure why github didn't add the comment to the review.
This suggestion removes the need for mutation and tracking the index. But I was also considering Isaac's suggestion to just load all of the projects, this would simplify the code. I am not sure how many billing projects we are expecting users to have.

```suggestion
          return h(ProjectDetail, {
            key: selectedName,
            billingProject: _.find({ projectName: selectedName }, billingProjects),
            billingAccounts,
            authorizeAndLoadAccounts,
            updateProjectAndReload: async () => _.flow(
              _.concat(_.reject({ projectName: selectedName }, billingProjects)),
              _.sortBy('projectName'),
              setBillingProjects
            )(await Ajax(signal).Billing.billingProject(selectedName))
          })
```

Reply:
Thank you for your suggestion. This code is much more complicated for the sake of avoiding assignment. I really don't think it's worth it - JavaScript is not a pure functional language and we seem bending over backwards to make it appear so.

Reply:
The billing project name can't change via the UI - in the case we can't access it, I've fallen back to just reloading everything